src/contracts/MsVault.h

```diff
@@ -0,0 +1,708 @@
+using namespace QPI;
+
+constexpr uint16 MAX_OWNERS = 32;
+constexpr uint16 INITIAL_MAX_VAULTS = 1024;
+constexpr uint16 MAX_VAULTS = INITIAL_MAX_VAULTS * X_MULTIPLIER;
+
+constexpr uint64 REGISTERING_FEE = 10000ULL;
+constexpr uint64 RELEASE_FEE = 1000ULL;
+constexpr uint64 RELEASE_RESET_FEE = 500ULL;
+constexpr uint64 HOLDING_FEE = 100ULL;
+
+constexpr uint16 VAULT_TYPE_QUORUM = 1;
+constexpr uint16 VAULT_TYPE_TWO_OUT_OF_X = 2;
+
+struct MSVAULT2
+{
+};
+
+struct MSVAULT : public ContractBase
+{
+    struct Vault
+    {
+        uint16 vaultType;
+        id vaultName;
+        array<id, MAX_OWNERS> owners;
+        uint16 numberOfOwners;
+        sint64 balance;
+        bit isActive;
+        array<uint64, MAX_OWNERS> releaseAmounts;
+        array<id, MAX_OWNERS> releaseDestinations;
+    };
+
+    struct MSVaultLogger
+    {
+        unsigned int _contractIndex;
```

Comment:
would it make sense to not mix between built in types an qpi types? (i mean use consequently qpi tpyes)